app/src/main/java/space/taran/arknavigator/ui/extra/ExtraLoader.kt

```diff
@@ -30,4 +33,35 @@ object ExtraLoader {
             else -> {}
         }
     }
+
+    fun loadInfo(
+        meta: ResourceMeta,
+        path: Path,
+        resourceId: TextView,
+        resourceName: TextView,
+        resourcePath: TextView,
+        resourceSize: TextView,
+        resourceResolution: TextView,
+        resourceDuration: TextView
+    ) {
+        // common resources for all file type
+        resourceId.text = String.format("Id\n%d", meta.id)
+        resourceName.text = String.format("Name\n%s", meta.name)
+        resourcePath.text = String.format("Path\n%s", path.absolutePathString())
+        resourceSize.text = String.format(
+            "Size\n%s", FileUtils.byteCountToDisplaySize(meta.size)
+        )
+        if (meta.kind == null) return
+        when (meta.kind) {
+            is ResourceKind.Video -> VideoExtraLoader.loadInfo(
+                meta.kind,
+                resourceResolution,
+                resourceDuration
+            )
+            else -> {
+                resourceResolution.makeGone()
+                resourceDuration.makeGone()
+            }
+        }
```

Comment:
What about other kinds?

Reply:
Resolution and duration only for video so we hide the both when kind is other than video.

Reply:
Also all common fields are managed already for all kinds.

Reply:
We also have a `Document` with page count and a `Link` with url, title, description.

Reply:
ohh my bad, I will update for this.

Reply:
Link description can be long, so it's better to add scroll view.